scaletest/workspacetraffic/run.go

```diff
@@ -186,20 +188,28 @@ type countReadWriter struct {
 	io.ReadWriter
 	bytesRead    atomic.Int64
 	bytesWritten atomic.Int64
+	metrics      *Metrics
+	labels       []string
 }
 
 func (w *countReadWriter) Read(p []byte) (int, error) {
+	start := time.Now()
 	n, err := w.ReadWriter.Read(p)
-	if err == nil {
+	w.metrics.ReadLatencySeconds.WithLabelValues(w.labels...).Observe(time.Since(start).Seconds())
+	if n > 0 {
 		w.bytesRead.Add(int64(n))
+		w.metrics.BytesRead.WithLabelValues(w.labels...).Add(float64(n))
 	}
 	return n, err
 }
 
 func (w *countReadWriter) Write(p []byte) (int, error) {
+	start := time.Now()
 	n, err := w.ReadWriter.Write(p)
-	if err == nil {
+	w.metrics.WriteLatencySeconds.WithLabelValues(w.labels...).Observe(time.Since(start).Seconds())
+	if n > 0 {
 		w.bytesWritten.Add(int64(n))
+		w.metrics.BytesWritten.WithLabelValues(w.labels...).Add(float64(n))
```

Comment:
nit: I'm curious if we can confirm a constant and consistent value (BytesWritten) with a unit test for metric.

Reply:
Yeah this is where it gets interesting. Each payload written includes a certain amount of overhead due to how `ReconnectingPTY` sends data. So sending 10 payloads of N bytes will incur more data transmission than 1 payload of N*10 bytes. 

That aside, I'd be worried about potential test flakiness. I can give it a try though!

Reply:
Oh, if you have any concerns, that is definitely not a must-have. Feel free to ignore this!

Reply:
I think I found a way to test the metrics sufficiently well.